### ISSUE TITLE
Download - System.NullReferenceException

### ISSUE THREAD

**Issue by vanicekv** [winsw/winsw#787]
**Summary**: The service will not start if it cannot connect to the download server 

## Steps to reproduce

Hello,

I use WinSW, including downloading file (.jar) from the server. If the server is available, everything will go fine. But if it is not or I enter, for example, an invalid server address, then the service will not start (it will crash immediately) and it will throw an error message, see below into the wrapper file.

```
2021-01-31 18:18:33,569 DEBUG - Starting WinSW in service mode
2021-01-31 18:18:33,616 INFO  - Downloading: https://something.com/none.jar to D:\none.jar. failOnError=False
2021-01-31 18:18:33,678 ERROR - Failed to download https://something.com/none.jar to D:\none.jar
System.AggregateException: One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at WinSW.Download.<PerformAsync>d__16.MoveNext()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object.
   at WinSW.Download.<PerformAsync>d__16.MoveNext()<---

2021-01-31 18:18:33,694 DEBUG - Completed. Exit code is 0
```

XML download config:
```
  <download from="https://something.com/none.jar" to="%BASE%\none.jar"
		  auth="basic" user="test" password="test" failOnError="false" />
```
I did not find a solution how to fix it, or what causes it. 

## Environment

* WinSW version: 2.10.3
* WinSW package type: .NET 4.6.1
* Windows version: Windows Server 2019 Datacenter/2016 Standard
* Wrapped executable and version: java jar

**Comment by cristianbojor** [comment]
Hello, did you manage to solve the issue?

For me the service fails every time it cannot download the files (no internet for example), even if I have set failOnError="false"

**Comment by vanicekv** [comment]
Hello,

Unfortunately, no. Still the same problem. I already thought that only I had this problem ... I use this solution on several servers and I have the same problem everywhere.

**Comment by cristianbojor** [comment]
Hello again,

I added a new issue with the information I found:
https://github.com/winsw/winsw/issues/814

For me it fails on every try, with both latest v3 and latest v2

**Comment by vanicekv** [comment]
Great, I hope the problem is resolved as soon as possible. I confirm that it also happens to me on both versions.

**Comment by nxtn** [comment]
This should have been fixed in 2.11.0 and 3.0.0-alpha.4.
